src/Microsoft.AspNetCore.WebUtilities/HttpResponseStreamWriter.cs

```diff
@@ -186,22 +220,43 @@ public override async Task WriteAsync(char[] values, int index, int count)
                 }
 
                 CopyToCharBuffer(values, ref index, ref count);
+                Debug.Assert(count == 0);
             }
         }
 
-        public override async Task WriteAsync(string value)
+        public override Task WriteAsync(string value)
         {
             if (_disposed)
             {
-                throw new ObjectDisposedException(nameof(HttpResponseStreamWriter));
+                return GetObjectDisposedTask();
             }
 
-            if (value == null)
+            var count = value.Length;
```

Comment:
This would now throw if `value` is `null`.

Reply:
fixed